src/ssh/sshConfig.ts

```diff
@@ -114,11 +137,20 @@ Include "gitpod/config"
                 throw new WrapError(`Could not read ssh config file at ${configPath}`, e);
             }
         }
-
-        const scopeRegex = new RegExp(`START GITPOD INTEGRATION.+Include "gitpod/config".+END GITPOD INTEGRATION`, 's');
-        if (!scopeRegex.test(content)) {
+        const oldContent = content;
+        const scopeRegex = new RegExp(`## START GITPOD INTEGRATION.+END GITPOD INTEGRATION`, 'sg');
+        const matchResult = content.match(scopeRegex);
+        if (matchResult) {
+            const matchContent = matchResult[0];
+            if (matchContent !== gitpodIncludeSection) {
+                content = content.replace(scopeRegex, gitpodIncludeSection);
+                // try to check and delete old file
+                tryDeleteOldMatch(matchContent, gitpodHeader);
+            }
+        } else {
             content = `${gitpodIncludeSection}\n\n${content}`;
-
+        }
+        if (content !== oldContent) {
```

Comment:
Should not we update SSH config only if we managed to copy our files? It does not make sense to include if we fail to copy?

Reply:
> Should not we update SSH config only if we managed to copy our files? It does not make sense to include if we fail to copy?

Right now it is.
<img width="893" alt="image" src="https://github.com/gitpod-io/gitpod-vscode-desktop/assets/20944364/7a1ce66f-2d07-4643-84ad-3d79980d4712">

Reply:
I meant `code_gitpod.d` first, if we cannto then there is no point to include it?

Reply:
It is a reason why we have broken include now for Chris in first place?

Reply:
I see, you mean to try to write `code_gitpod.d/config` first ? OK, make sense